### ISSUE TITLE
APIs

### ISSUE THREAD

**Issue by WhoSoup** [pegnet/pegnet#223]
I noticed a few issues floating around that were about API sources, namely #204 and and #203. I wanted to open a broader issue that we currently face: the free plans are not viable. We need alternatives.

First of all, we need a maximum of **~4320 requests a month** or **144 a day** with an update granularity of **10 minutes**.

** CurrencyLayer (APILayer) **
The free plan updates **daily** with **250 req/mo**, both of which is not enough. The cheapest plan to satisfy our requirements is the professional plan at **$40/mo**.

** CoinCap **
Updates **live** with no apparent rate limit. There is no paid option. This seems to suit our needs.

** CoinMarketCap **
The free plan updates **live** with **10,000 req/mo** (333 a day). This seems to suit our needs.

** ExchangeRatesAPI **
Updates **daily** with no request limit. This does not suit our needs.

** Kitco **
Updates seem to be **<10m** but only while the markets are open. Unknown requests due to web scraping.

** OpenExchangeRates **
Updates **hourly** with **1,000 req/mo**, both of which fail our needs. The cheapest plan with a sub-10 granularity is the "unlimited" plan at **$97/mo**.

That means we are only left with three workable free APIs that suit our needs: CoinCap (crypto only), CoinMarketCap (crypto only), and Kitco (precious metals only). We currently do not have a free live source of currency data, it seems.

**Comment by PaulBernier** [comment]
In additions to the points above it was discussed on Discord that the default data sources should have Terms of Service allowing the publishing of their data publicly on the blockchain, which is not the case for most of our current data sources. The reason is that it's not reasonable to expect the average solo miner to go negotiate a deal with those data providers and as such the default config should protect the user from involuntarily breaking ToS (it wouldn't be great for the image of the project either). The miner would then be free to change to alternate data sources while being made aware (s)he has to pay particular attention to the ToS of the provider (s)he wish to use.

**Comment by WhoSoup** [comment]
> default data sources should have Terms of Service allowing the publishing of their data publicly on the blockchain

yeah. this is something where a lawyer could help save us a lot of headache. publishing "their data" could just mean that they don't want someone to run a man-in-the-middle service but if we are just using partial data (just the prices not the metadata) then it might be okay. I don't know the answer to that.

**Comment by WhoSoup** [comment]
Comparison of the price of USD on CoinCap (samples every 60 seconds) vs CoinMarketCap (sampled every 600 seconds)

![image](https://user-images.githubusercontent.com/34172041/62852625-3c353580-bcea-11e9-915b-f3fc9241e995.png)

**Comment by tmlee** [comment]
Hey @WhoSoup have you also considered the CoinGecko API (https://www.coingecko.com/api) ?
We do have direct conversion to Gold and Silver spot price alongside the national currencies and such. API is publicly available and no keys required.

Here is one example of the call , there are more in the docs too to choose from depending on your use case
https://api.coingecko.com/api/v3/simple/price?ids=bitcoin,ethereum&vs_currencies=xau,xag,usd

**Comment by WhoSoup** [comment]
> Hey @WhoSoup have you also considered the CoinGecko API (https://www.coingecko.com/api) ?

Thanks for letting us know about this
